### PR TITLE
Ignoe derp.yaml, don't panic in Serve()

### PR DESCRIPTION
**Pull request by negbie**
If there is no specifc reason to panic I think it's better to just return the error.

### CODE REVIEW

**Comment by negbie** [comment]
@kradalby Do you know why the integration-test for the second pr fails? It adds only more logging.

**Comment by kradalby** [comment]
They are a bit flaky, I think they are a bit more resource hungry than actions wants to so they might get in an awkward state.

Reruns usually helps...

**Comment by negbie** [comment]
Makes sense. Thanks for the info!

File: app.go [hunk 1]
```diff
@@ -418,12 +418,12 @@ func (h *Headscale) Serve() error {
 
 	err = h.ensureUnixSocketIsAbsent()
 	if err != nil {
-		panic(err)
+		return err
```
Comment (kradalby) [inline app.go:421]: I think this is sensible,

But could you wrap these errors to give the user more context, when we are first at improving :)

So:
```suggestion
		return fmt.Errorf("description: %w", err)
```

File: app.go [hunk 2]
```diff
@@ -418,12 +418,12 @@ func (h *Headscale) Serve() error {
...
 	err = h.ensureUnixSocketIsAbsent()
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("unable to remove old socket file: %w", err)
 	}
 
 	socketListener, err := net.Listen("unix", h.cfg.UnixSocket)
 	if err != nil {
-		panic(err)
+		return err
```
Comment (kradalby) [inline app.go:426]:
```suggestion
		return fmt.Errorf("failed to set up gRPC socket: %w", err)
```

File: app.go [hunk 3]
```diff
@@ -441,7 +441,7 @@ func (h *Headscale) Serve() error {
 
 	networkListener, err := net.Listen("tcp", h.cfg.Addr)
 	if err != nil {
-		panic(err)
+		return err
```
Comment (kradalby) [inline app.go:444]:
```suggestion
		return fmt.Errorf("failed to bind to TCP address: %w", err)
```

**Review by kradalby — APPROVED** [review]

**Pull request merged**